Wrappers/Python/test/test_ring_processor.py

```diff
@@ -0,0 +1,172 @@
+
+#   Copyright 2017 UKRI-STFC
+#   Copyright 2017 University of Manchester
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+        
+
+from ccpi.processors import RingRemover
+from ccpi.framework import ImageData, ImageGeometry, AcquisitionGeometry
+from ccpi.astra.operators import AstraProjectorSimple
+
+import tomophantom
+from tomophantom import TomoP2D       
+import os
+import numpy as np
+import unittest
+from ccpi.io import NEXUSDataReader
+
+class TestRingProcessor(unittest.TestCase):
+    
+    def test_2D_demo_ring(self):
+        
+        print("Start 2D ring removal in simulated sinogram")
+    
+        model = 1 # select a model number from the library
+        N = 512 # set dimension of the phantom
+        path = os.path.dirname(tomophantom.__file__)
+        path_library2D = os.path.join(path, "Phantom2DLibrary.dat")
+        
+        phantom_2D = TomoP2D.Model(model, N, path_library2D)    
+        data = ImageData(phantom_2D)
+        ig = ImageGeometry(voxel_num_x=N, voxel_num_y=N, voxel_size_x = 0.1, voxel_size_y = 0.1)
+        
+        # Create acquisition data and geometry
+        detectors = N
+        angles = np.linspace(0, np.pi, 120)
+        ag = AcquisitionGeometry('parallel','2D',angles, detectors, pixel_size_h = 0.1)
+            
+        Aop = AstraProjectorSimple(ig, ag, 'cpu')
+        sin = ag.allocate()
+        Aop.direct(data, out = sin)
+        
+        sin_stripe = 0*sin
+    #        sin_stripe = sin
+        tmp = sin.as_array()
+        tmp[:,::20]=0
+        sin_stripe.fill(tmp)
+                
+        ring_removal = RingRemover(4, "db25", 20, info = True)
+        ring_removal.set_input(sin_stripe)
+        ring_recon = ring_removal.get_output()
+        
+        # load ring processed result sinogram 2D 
+        reader = NEXUSDataReader()
+        reader.set_up(nexus_file = os.path.join(os.getcwd(), 'result_sinogram_2D_ring_remover.nxs'))
+        tmp = reader.load_data()      
+        
+        print("Check ring remover sinogram 2D")
+        np.testing.assert_array_equal(tmp.as_array(), ring_recon.as_array()) 
+        print("Test passed\n")        
+                    
+    
+    def test_3D(self):
+        
+        print("Start 3D ring removal in real data")
+        
+        # Load data --> sirt after flat and after ring
+        read_sinogram3D = NEXUSDataReader()
+        read_sinogram3D.set_up(nexus_file = 'Sinogram_Rock_sample_3D.nxs')
```

Comment:
Needs to download the data as in [here](https://github.com/vais-ral/CCPi-Framework/blob/master/Wrappers/Python/test/test_NexusReader.py#L35)